sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java

```diff
@@ -22,42 +22,50 @@
 package io.crate.analyze;
 
 import io.crate.action.sql.SessionContext;
-import io.crate.data.Row;
-import io.crate.metadata.PartitionName;
+import io.crate.analyze.expressions.ExpressionAnalysisContext;
+import io.crate.analyze.expressions.ExpressionAnalyzer;
+import io.crate.analyze.relations.FieldProvider;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.Functions;
 import io.crate.metadata.RelationName;
 import io.crate.metadata.Schemas;
 import io.crate.metadata.doc.DocTableInfo;
 import io.crate.metadata.table.Operation;
 import io.crate.sql.tree.AlterTable;
 import io.crate.sql.tree.AlterTableRename;
-import io.crate.sql.tree.Assignment;
 import io.crate.sql.tree.Expression;
-import io.crate.sql.tree.Table;
-import org.elasticsearch.cluster.metadata.IndexMetaData;
-import org.elasticsearch.common.settings.Settings;
+import io.crate.sql.tree.ParameterExpression;
 
-import javax.annotation.Nullable;
 import java.util.List;
+import java.util.function.Function;
 
 import static io.crate.analyze.BlobTableAnalyzer.tableToIdent;
 
 public class AlterTableAnalyzer {
 
     private final Schemas schemas;
+    private final Functions functions;
 
-    AlterTableAnalyzer(Schemas schemas) {
+    AlterTableAnalyzer(Schemas schemas, Functions functions) {
         this.schemas = schemas;
+        this.functions = functions;
     }
 
-    public AlterTableAnalyzedStatement analyze(AlterTable node, Row parameters, SessionContext sessionContext) {
-        Table<Expression> table = node.table();
-        DocTableInfo docTableInfo = (DocTableInfo) schemas.resolveTableInfo(table.getName(), Operation.ALTER_BLOCKS,
-            sessionContext.searchPath());
-        PartitionName partitionName = createPartitionName(table.partitionProperties(), docTableInfo, parameters);
-        TableParameters tableParameters = getTableParameterInfo(table, partitionName);
-        TableParameter tableParameter = getTableParameter(node, parameters, tableParameters);
-        maybeRaiseBlockedException(docTableInfo, tableParameter.settings());
-        return new AlterTableAnalyzedStatement(docTableInfo, partitionName, tableParameter, table.excludePartitions());
+    public AnalyzedAlterTable analyze(AlterTable<Expression> node,
+                                               Function<ParameterExpression, Symbol> convertParamFunction,
+                                               CoordinatorTxnCtx txnCtx) {
+        var exprAnalyzerWithFieldsAsString = new ExpressionAnalyzer(
+            functions, txnCtx, convertParamFunction, FieldProvider.FIELDS_AS_LITERAL, null);
+        var exprCtx = new ExpressionAnalysisContext();
```

Comment:
minor: we probably do not need this initialization

Reply:
why not?

Reply:
it is accessed only once in the `analyze` method

Reply:
yes but inside a function, otherwise a new context would be created on each function call. or do I miss something?